### ISSUE TITLE
react/prop-types false positive when component is exported to a different name

### ISSUE THREAD

**Issue by bdwain** [yannickcr/eslint-plugin-react#2051]
I have a component that is giving me false positives saying that a proptype is not defined.

`'children' is missing in props validation  react/prop-types`

Here's an example

```js
import React from 'react';
import PropTypes from 'prop-types';
import makeHoc from './make-hoc';
import selectors from './selectors';

class Foo extends React.Component {
  render() {
    return this.props.children;
  }
}

Foo.propTypes = {
  children: PropTypes.element.isRequired
};

export const Unconnected = Foo; //this line causes the issue
export default makeHoc(
  Foo,
  selectors
);
```

and it works if I change commented line above to this

```js
export { Foo as Unconnected };
```

At first I thought #1958 was related but on second look it doesn't seem to be. I also tried downgrading to 7.10.0 and that didn't help.

I don't see anything wrong with that syntax, but please point out if I'm missing something. Thanks!

**Comment by ljharb** [comment]
Certainly the `as` version is better, since it doesn't create an additional binding inside your module, but the `=` version should still work.

**Comment by bdwain** [comment]
Agreed. But figured it was still a bug so I'd report it.

**Comment by alexzherdev** [comment]
Hmm I pasted this entire example as a test for 7.11.1 and it passed

**Comment by alexzherdev** [comment]
oh cool, according to the build in #2068 this does fail on ESLint 5.

**Comment by alexzherdev** [comment]
Also, this has been filed before in #1957
